src/replit_river/codegen/typing.py

```diff
@@ -145,6 +162,12 @@ def work(
         raise ValueError("Incoherent state when trying to flatten unions")
 
 
+def normalize_special_chars(value: str) -> str:
+    for char in SPECIAL_CHARS:
+        value = value.replace(char, "_")
```

Comment:
this can cause name collisions (e.g. ".dark" and "-dark" would collide)

Reply:
Is erroring an acceptable action if two names would collide or do you think using a more unique normalization (like using a char code as bri suggested) would be better?

Reply:
or just phoneticizing (e.g. _ -> underscore, . -> dot) i think erroring is fine too